cloudinit/stages.py

```diff
@@ -953,23 +956,38 @@ def should_run_on_boot_event():
 
 
 def read_runtime_config():
-    return util.read_conf(RUN_CLOUD_CONFIG)
+    try:
+        return util.read_conf(RUN_CLOUD_CONFIG)
+    except OSError as e:
+        if e.errno == errno.EACCES:
+            LOG.warning(
+                "REDACTED config part %s for non-root user",
+                RUN_CLOUD_CONFIG,
+            )
+        raise
 
 
 def fetch_base_config():
-    return util.mergemanydict(
-        [
-            # builtin config, hardcoded in settings.py.
-            util.get_builtin_cfg(),
-            # Anything in your conf.d or 'default' cloud.cfg location.
-            util.read_conf_with_confd(CLOUD_CONFIG),
-            # runtime config. I.e., /run/cloud-init/cloud.cfg
-            read_runtime_config(),
-            # Kernel/cmdline parameters override system config
-            util.read_conf_from_cmdline(),
-        ],
-        reverse=True,
-    )
+    config_loaders = [
+        # builtin config, hardcoded in settings.py.
+        util.get_builtin_cfg,
```

Comment:
While I appreciate the sentiment in the event we get permissions errors, the following callables will not generate EACCES errors so it might be over-engineering to cope with scenarios which won't raise EACCESS:
- `get_builtin_cfg` (we are sourcing a static variable in the python code)
- `read_runtime_config` pulls from /run/cloud-init which we set at world-readable
- `read_conf_from_cmdline` pulls from /proc/cmdline which is also world-readable


The only function/callable that could generate that EACCES is read_conf_with_confd, so I don't think we need to add the layer of test/except handling here in a for loop if we handle this warning down in `read_conf_with_confd`

Reply:
After previous modifications, this is definitely not needed. Removed.